src/clojure/nrepl/misc.clj

```diff
@@ -60,3 +60,20 @@
         (require (symbol (namespace sym)))
         (resolve sym)
         (catch Exception _))))
+
+(defmacro with-session-classloader
```

Comment:
You'll have to make sure that hotloading still works. I'm always afraid we might break it when fiddling with the classloader, as it doesn't really have any tests.

Reply:
hotloading as in what Pomegranate does?

Dynamic loader capabilities is fairly separate from sideloading, if that's what you are concerned about. This macro is from the changes made in #162, which has its origins in unrepl. Both sideloading + dynamic-loader while sideloading are covered in tests.

I was looking at #113 : is that the kind of breakage you are concerned about? Maybe the the thing to do is to add more tests that we are concerned about?

Reply:
> hotloading as in what Pomegranate does?

Yep. It didn't work with nREPL until we did some classloader modifications a while ago, so I'm always wary of us breaking this again. 

> I was looking at #113 : is that the kind of breakage you are concerned about? Maybe the the thing to do is to add more tests that we are concerned about?

Yeah, we should probably tackle this at some point. Initially we skipped the tests because of the many moving pieces there (hotloading library, build tool).

Reply:
I've made the macro a bit safer. If there's no `:classloader` under (meta session), then it's pretty much a `do` block.

Currently, only the sideloader sets `:classloader`, so it's quite unlikely to make any breaking changes. I'd be happy to chat more about tests/classloader etc. separately though?